library/kotlin/src/io/envoyproxy/envoymobile/Client.kt

```diff
@@ -0,0 +1,32 @@
+package io.envoyproxy.envoymobile
+
+import java.nio.ByteBuffer
+
+interface Client {
+  fun startStream(request: Request, streamCallback: StreamCallback): StreamEmitter
+}
+
+interface StreamCallback {
+  fun onHeaders(statusCode: Int, headers: Map<String, List<String>>)
+  fun onData(byteBuffer: ByteBuffer, endStream: Boolean)
+  fun onMetadata(metadata: Map<String, List<String>>, endStream: Boolean)
+  fun onTrailers(trailers: Map<String, List<String>>)
+  fun onError(error: Error)
+  fun onCanceled()
+  fun onCompletion()
+}
+
+interface StreamEmitter {
+  fun isActive(): Boolean
+  fun sendData(byteBuffer: ByteBuffer, endStream: Boolean): StreamEmitter
```

Comment:
`sendData` and `sendMetadata` should throw if they fail to succeed (i.e., `isActive() = false`), right?

Reply:
I don't think the exception case when `isActive() == false` needed to be declared for the compiler

We can declare that exception, the case when we are unable to send data